tamr_unify_client/attribute/type.py

```diff
@@ -32,8 +30,10 @@ def inner_type(self):
     @property
     def attributes(self):
         """:type: list[:class:`~tamr_unify_client.attribute.subattribute.SubAttribute`]"""
+        from tamr_unify_client.attribute.subattribute import SubAttribute
```

Comment:
This is for my own education, but why move the import from the file level to the function level? @pcattori

Reply:
i think this is mostly to not run all that import logic unless it's needed

Reply:
`Attribute` and `SubAttribute` both depend on each other. If they each declare the other as imports at the file level, this will cause a [cyclic dependency](http://effbot.org/zone/import-confusion.htm#circular-imports). 

Previously, we got around this by having `SubAttribute` import `Attribute` at the function level, but conceptually `SubAttribute` depends much more heavily on `Attribute` than the other way around. So I changed this to reflect that. Specifically, now `SubAttribute` imports `Attribute` at the file level since it fundamentally depends on `Attribute`, whereas `Attribute` only depends on `SubAttribute` in 1 context (this one).

Reply:
to be clear: the code won't run successfully with cyclic imports. it will crash. so this is a correctness thing, not a perf thing